### PR TITLE
ci: Add more context to the Cypress docker environment

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Passes more env data to cypress in docker
- This should allow us to better link cypress runs to Github PRs

### CODE REVIEW

**Review by jdamore-linode — APPROVED** [review]
Awesome work @bnussman-akamai!

File: docker-compose.yml [hunk 1]
```diff
@@ -6,16 +6,19 @@ x-e2e-env:
   BUILD_ID: "${GIT_BRANCH}-${BUILD_NUMBER}"
   BUILD_URL: ${BUILD_URL}
   BUILD_NUMBER: ${BUILD_NUMBER}
-  COMMIT_INFO_BRANCH: ${GIT_BRANCH}
+  COMMIT_INFO_BRANCH: ${CHANGE_BRANCH}
```
Comment (hkhalil-akamai) [inline docker-compose.yml:9]: What's the difference between `GIT_BRANCH` and `CHANGE_BRANCH`?

Comment (bnussman-akamai) [reply docker-compose.yml:9]: `GIT_BRANCH` was giving us the value `PR-8970` but we want the actual branch name. @jdamore-linode and I found that `CHANGE_BRANCH` gives us a more descriptive branch name. For this PR `CHANGE_BRANCH` is `ci/add-more-cypres-context`

**Comment by bnussman-akamai** [comment]
I was able to remove `COMMIT_INFO_AUTHOR` because after a change Joe made last night, cypress seems to be able to infer this value without this PR.

**Review by cpathipa — APPROVED** [review]

**Pull request merged**